watcher/src/bin/avr_export.rs

```diff
@@ -0,0 +1,158 @@
+        parse(from_os_str),
+        env = "MC_WATCHER_DB"
+    )]
+    pub watcher_db: PathBuf,
+
+    /// Path for the avr-history.toml & avr-history.json bootstrap files to be
+    /// written.
+    #[clap(
+        long,
+        default_value = "",
+        parse(from_os_str),
+        env = "MC_AVR_HISTORY_PATH"
+    )]
+    pub avr_history: PathBuf,
+}
+
+fn main() {
+    let (logger, _global_logger_guard) = create_app_logger(o!());
+
+    let mut config = Config::parse();
+    config.avr_history.set_file_name("avr_history");
+    let watcher_db =
+        WatcherDB::open_ro(&config.watcher_db, logger).expect("Failed opening watcher db");
+    let mut avr_reports = AvrConfig::default();
+
+    // Get all of the latest synced blocks
+    let last_synced_blocks = watcher_db.last_synced_blocks().unwrap();
+
+    // Find where the signing keys changed and fetch the AVRs for those keys
+    for (tx_src_url, max_block_index) in last_synced_blocks.iter() {
+        let max_block_count = max_block_index.map_or_else(|| 0, |idx| idx + 1);
+        let mut cur_start_index = 0;
+        let mut cur_end_index = 0;
+        let mut cur_signer = None;
+        for block_index in 0..max_block_count {
+            let signer = match watcher_db.get_block_data(tx_src_url, block_index) {
+                Ok(block_data) => block_data.signature().map(|sig| *sig.signer()),
+                Err(WatcherDBError::NotFound) => None,
+                Err(err) => {
+                    panic!(
+                        "Failed getting block {}@{}: {:?}",
+                        cur_start_index, tx_src_url, err
+                    );
+                }
+            };
+
+            if signer == cur_signer {
+                cur_end_index += 1;
+            } else {
+                let record = generate_avr_record(
+                    &watcher_db,
+                    tx_src_url,
+                    cur_start_index,
+                    cur_end_index,
+                    &signer,
+                );
+                avr_reports.avr_records.push(record);
+                cur_start_index = block_index;
+                cur_end_index = block_index;
+                cur_signer = signer;
+            }
+        }
+    }
+
+    // Export the AVR history to toml and write it to disk
+    let avr_history_toml = toml::to_string_pretty(&avr_reports).unwrap();
+    let avr_history_json = serde_json::to_string_pretty(&avr_reports).unwrap();
+    println!("Historical AVR Results: \n\n{}", avr_history_toml);
+    config.avr_history.set_extension("toml");
+    fs::write(&config.avr_history, avr_history_toml).unwrap();
+    config.avr_history.set_extension("json");
+    fs::write(&config.avr_history, avr_history_json).unwrap();
+}
+
+// Replace the archive prefix with the consensus prefix (mc://)
+fn create_responder_id(url: &Url) -> ResponderId {
+    if url.scheme() == "https" {
+        return ResponderId(format!("mc:/{}", url.path()));
```

Comment:
1. I was a little confused on this, it could be switched to host:port. The problem is within most of the watcher databases I've received, the "ResponderId" is the S3 or Mobilecoin ledger backfill urls that represent the responders. They do however have the ResponderIDs within those urls, so we can grab those.
2. Switching to host/port will be take care of the uri scheme.

Reply:
We won't be able to get the port from the current WatcherDB but we can get the host.

Reply:
This likely means that once the file is generated with whatever responder id ends up in it, a human needs to go and edit it. Since this only happens once I believe that is acceptable.

Reply:
Even we even have that information, I'm not entirely sure we do